README.md

```diff
@@ -70,15 +119,279 @@ solidity: |-
+  - to `/fasthook` to complete immediately when the message is sent to Kafka
+- Get an _immediate_ receipt
+   - No need to wait many seconds for a block to be cut
+   - No need to have complex retry handling
+
+This is ideal for Applications, Integration-as-a-Service (IaaS) platforms, and built-in SaaS integrations that support Webhooks for event submission.
+
+However, SaaS administrators and end-users still need to be able to track down what happened for a particular transaction, to diagnose problems, and correlate details of the Ethereum transaction with the business event.
+
+So kaleido-io/ethereum comes with a built-in receipt store, backed by MongoDB. It listens reliably for the replies over Kafka, and inserts each of them into the Database.
+
+It provides a trivially simple REST API:
+- `GET` `/reply/a789940d-710b-489f-477f-dc9aaa0aef77` to look for an individual reply
+- `GET` `/replies` to list the replies
+  - Ordered by time _received_ (not the order submitted) - listing the newest first
+  - `limit` and `skip` query parameters can be used to paginate the results
+
+A capped collection can be used in MongoDB to limit the storage. For example to store only the last 1000 replies received.
+
+### Scale and message ordering - nonce management
+
+The transaction pooling/execution logic within an Ethereum node is based upon the concept of a `nonce`, which must be incremented exactly once each time a transaction is submitted from the same Ethereum address. There can be no gaps in the nonce values, or messages build up in the `queued transaction` pool waiting for the gap to be filled. This allows for deterministic ordering of transactions sent by the same sender.
+
+The management of this `nonce` pushes complexity back to the application tier - especially for horizontally scaled Enterprise applications sending many transactions using the same sender. By using an ordered Messaging stream to submit messages into the Ethereum network, many applications are able to delegate this complexity to kaleido-io/ethconnect.
+
+The kaleido-io/ethconnect bridge contains all the logic necessary to communicate with the node to determine the next nonce, and also to cope with multiple requests being in flight within the same block, for the same sender (including [with IBFT](https://github.com/ethereum/EIPs/issues/650#issuecomment-360085474)).
+
+If a sender needs to achieve exactly-once delivery of transactions (vs. at-least-once) it is still necessary to allocate the nonce within the application and pass it into kaleido-io/ethconnect in the payload.  This allows the sender to control allocation of nonces using its internal state store / locking.
+
+Given many Enterprise scenarios involve writing hashed proofs of completion of an off-chain transaction to a shared ledger (vs. performing the actual transaction on the chain), at-least-once delivery is sufficient in a wide range of cases. Smart Contracts can simply re-store the proof.
+
+## Why Kafka?
```

Comment:
I would put this in a FAQ in the Wiki.

Reply:
Fair suggestion that we've cluttered the doc, but I've found wikis very painful in Git in the past. A bit of an information black-hole. I think this is pretty critical information for a new user, because one of the OSS AMQP providers (or even IBM MQ) might seem the obvious choice to many readers